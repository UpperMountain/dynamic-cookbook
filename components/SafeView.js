import React from "react";
import { SafeAreaView, Platform, StatusBar } from "react-native";

const SafeView = ({ children, style, ...rest }) => (
  <SafeAreaView
    style={{
      ...style,
      flex: 1,
      paddingTop: Platform.select({
        android: StatusBar.currentHeight,
        ios: 0
      })
    }}
    {...rest}
  >
    {children}
  </SafeAreaView>
);

export default SafeView;
