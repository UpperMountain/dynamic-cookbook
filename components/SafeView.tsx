import React from "react";
import { ViewProps } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

interface Props extends ViewProps {
  children: React.ReactNode;
  style?: ViewProps["style"];
}

const SafeView = ({ children, style, ...rest }: Props) => (
  <SafeAreaView
    style={[
      {
        flex: 1
      },
      style
    ]}
    {...rest}
  >
    {children}
  </SafeAreaView>
);

export default SafeView;
