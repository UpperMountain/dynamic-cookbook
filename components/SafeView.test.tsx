import React from "react";
import { Text } from "react-native";
import SafeView from "./SafeView";
import renderer from "react-test-renderer";

it("should render without exploding", () => {
  renderer
    .create(
      <SafeView>
        <Text>testing</Text>
      </SafeView>
    )
    .unmount();
});

it("should pass snapshot tests", () => {
  const cmp = renderer.create(
    <SafeView>
      <Text>testing</Text>
    </SafeView>
  );
  expect(cmp.toJSON()).toMatchSnapshot();
  cmp.unmount();
});

it("should consistently handle a style prop", () => {
  const cmp = renderer.create(
    <SafeView style={{ backgroundColor: "red" }}>
      <Text>testing</Text>
    </SafeView>
  );
  expect(cmp.toJSON()).toMatchSnapshot();
  cmp.unmount();
});
