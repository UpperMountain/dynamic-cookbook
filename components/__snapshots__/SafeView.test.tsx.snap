// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should consistently handle a style prop 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 0,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text>
    testing
  </Text>
</RCTSafeAreaView>
`;

exports[`should pass snapshot tests 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
        "paddingTop": 0,
      },
      undefined,
    ]
  }
>
  <Text>
    testing
  </Text>
</RCTSafeAreaView>
`;
